### PR TITLE
Entity diagram improvements

### PR DESCRIPTION
**Pull request by chowington**
- Combine the two entity diagram components into one
- Implement precise shading
- Fix mini diagram tooltips

Note that the new unified `EntityDiagram` component has all of the latest improvements; the others will not be updated after this PR and may be removed later.

### CODE REVIEW

**Review by dmfalke — CHANGES_REQUESTED** [review]
This looks great!

A few requests:
1. Get rid of `ExpandedDiagram.tsx` and `MiniDiagram.tsx`. I think the combined one should be sufficient.
2. I see some React key warnings being logged in the console. Can you address these? See https://reactjs.org/link/warning-keys.

**Review by bobular — CHANGES_REQUESTED** [review]
Looks good Connor.

Just the duplicated type definition that needs attention, and I recommend merging #86 so the story can be adjusted on the fly by testers.

Could you also add a bit of padding/margin around the diagram (as per Steve's suggestion) and maybe also drop-shadow?

File: src/EntityDiagram/Types.ts [hunk 1]
```diff
@@ -24,17 +24,10 @@ export interface StudyData {
...
 
-export type ShadingValue = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
-
-export interface ShadingValues {
-  /** The portion of the rectangle to shade (0: 0%, 1: 10%, ... 10: 100%) */
-  value: ShadingValue;
-  // color?: string;  // Not implemented yet
-}
-
 export interface ShadingData {
```
[the remark this answers is not in the document]
Comment (chowington) [reply src/EntityDiagram/Types.ts:27]: > 
> 
> This looks great!
> 
> A few requests:
> 
>     1. Get rid of `ExpandedDiagram.tsx` and `MiniDiagram.tsx`. I think the combined one should be sufficient.
> 
>     2. I see some React key warnings being logged in the console. Can you address these? See https://reactjs.org/link/warning-keys.

Thanks Dave! The only reason I didn't scrap the two components was for the animation; I know that was something we wanted for switching between them. I can delete them and look that up in the history if I need to (unless I forget it's there :).

I also remember seeing some unique key warnings as well. After a quick check it wasn't clear where exactly they were coming from, but I'll look at it more closely.

[the remark this answers is not in the document]
Comment (dmfalke) [reply src/EntityDiagram/Types.ts:27]: Yeah, that animation doesn't seem to be complete, so it might be better to just start new when we know we want it.

**Comment by chowington** [comment]
> 
> 
> Looks good Connor.
> 
> Just the duplicated type definition that needs attention, and I recommend merging #86 so the story can be adjusted on the fly by testers.
> 
> Could you also add a bit of padding/margin around the diagram (as per Steve's suggestion) and maybe also drop-shadow?

Thanks Bob!

Yeah, the duplicated type definition was so `EntityDiagram` wouldn't be dependent on `Types`, as the `Types` definitions would disappear along with the two separate ED components that needed them. I'll just finish the job and get rid of all of that according to Dave's request 👍 

I'll talk about the padding on Slack. Drop shadow exists on my branch along with controls 👍

**Review by dmfalke — APPROVED** [review]
These look good to me. I'll let @bobular make the final approval and merge.

**Comment by bobular** [comment]
OK, I'll merge now.  Thanks!
Will you do another PR for the controls, drop-shadow etc?

**Pull request merged**